schema_based_on_diagram.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE patients (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255),
+  date_birth DATE
+);
+
+CREATE TABLE medical_histories (
+  id SERIAL PRIMARY KEY,
+  admit_at TIMESTAMP,
+  patient_id INT,
+  status VARCHAR(50),
+  FOREIGN KEY (patient_id) REFERENCES patients (id)
+);
+
+CREATE TABLE invoices (
+  id SERIAL PRIMARY KEY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history_id INT,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
+CREATE TABLE invoice_items (
+  id SERIAL PRIMARY KEY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treament_id INT,
+  FOREIGN KEY (invoice_id) REFERENCES invoices (id),
+  FOREIGN KEY (treament_id) REFERENCES treaments (id)
+);
+CREATE TABLE treaments (
+  id SERIAL PRIMARY KEY,
+  type INT,
+  name VARCHAR(225),
+  medical_history_id INT,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
```

Comment:
- [ ] [optional] @massazoum and @harshi0102 great work implementing the changes requested, however, you forgot to create all "join table" for the many-to-many relationships, between `medical_histories` and `treatments` , also you dont have a foreign key on the treatments table the relationship is suppose to be covered using the join table, kindly consider implementing the join table and add indexes on its foreign keys
```
  CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```